src/panels/video-player/VideoPlayer.tsx

```diff
@@ -4,18 +4,52 @@ import 'aws-iot-twinmaker-grafana-utils/dist/index.css';
 import { getStyles } from './styles';
 import { VideoPlayerPropsFromParent } from './interfaces';
 import { auto } from '@popperjs/core';
+import { getTemplateSrv, locationSearchToObject } from '@grafana/runtime';
+import { getUrlTempVarName, tempVarSyntax } from 'common/variables';
+import { useLocation } from 'react-router-dom';
 
 export const VideoPlayer = (props: VideoPlayerPropsFromParent) => {
   const styles = getStyles();
+  const templateSrv = getTemplateSrv();
+
+  const { search } = useLocation();
 
   // Automatically update the video player when the related parameters change
   const setVideoPlayer = useCallback(() => {
+    // Get variables from the URL
+    const queryParams = locationSearchToObject(search || '');
+    let entityId = queryParams[getUrlTempVarName(props.options.entityId)] as string;
+    let componentName = queryParams[getUrlTempVarName(props.options.componentName)] as string;
+    let kvsStreamName = queryParams[getUrlTempVarName(props.options.kvsStreamName)] as string;
+
+    // Double check default variable values if url is empty
+    if (!entityId) {
+      const entityIdVar = tempVarSyntax(props.options.entityId);
```

Comment:
The following 3 if blocks has same code - will be better to separate out in a function and avoid duplication.

Reply:
Yes, will condense this in the next commit